app/cdap/components/ConfigurationGroup/PropertyRow/index.tsx

```diff
@@ -83,6 +83,8 @@ const WIDGET_CATEGORY = 'widget-category';
 
 interface IState {
   isMacroTextbox: boolean;
+  prevValue: string;
+  prevMacroValue: string;
```

Comment:
Persisting the previously entered macro was not part of the original issue. I figured we could do it since we're persisting non-macro entries. Let me know if it makes sense.